aws/validators.go

```diff
@@ -54,6 +54,22 @@ func validateTypeStringNullableBoolean(v interface{}, k string) (ws []string, es
 	return
 }
 
+// validateTypeStringNullableFloat provides custom error messaging for TypeString floats
+// Some arguments require a floating point value or an unspecified, empty field.
+func validateTypeStringNullableFloat(v interface{}, k string) (ws []string, es []error) {
+	value, ok := v.(string)
+	if !ok {
+		es = append(es, fmt.Errorf("expected type of %s to be string", k))
+		return
+	}
+
+	if _, err := strconv.ParseFloat(value, 64); err != nil {
```

Comment:
Should there be a check for empty string in here? May help to do a little unit testing on this as there are some edges. This seems to error if empty string.

https://play.golang.org/p/iBa1Pq5V2tt